### PR TITLE
Added action check for different source and target branches

### LINKED ISSUE #1704: GitHub Action: Fail when the PR source and target branches are the same
**Is your feature request related to a problem? Please describe.**

1. We frequently have to close PRs when the source and target PR branches are the same. This means that our reviewers will not be able to checkout the changes and test on their systems using this methodology which will cause a rejection error:
    1. https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/checking-out-pull-requests-locally
1. For example if a the author of the PR is working on the `develop` branch in their local repository and submits a PR to this repository's `develop` branch, then the PR reviewer will not be able to use the procedure in the link to review the operation of the code first hand.

**Describe the solution you'd like**

1. A GitHub action that is part of the code quality checks section that fails when the source and target branches are the same
    1. https://github.com/PalisadoesFoundation/talawa-admin/blob/develop/.github/workflows/pull-request.yml
1. The YAML block added must clearly indicate that: 
    1. it's testing whether the source and destination branches are different. 
    2. the names of the source and destination branch names with explicatory text, possibly using one or more echo commands

**Describe alternatives you've considered**

- N/A

**Approach to be followed (optional)**

- N/A

**Additional context**

- Here is a screen shot of an example where the two branches are the same. (`develop`)
    image

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by aarishshahmohsin**
**What kind of change does this PR introduce?**

This PR adds a github action check for the source and target branches of a pull request to be different.

**Issue Number:**

Fixes #1704 

**Did you add tests for your changes?**

NA

**Snapshots/Videos:**

NA

**If relevant, did you update the documentation?**

NA

**Summary**

Now any PR will not pass the tests if the source and target branches are the same.

**Does this PR introduce a breaking change?**

No

**Other information**

NA

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

**Comment by Cioppolo14** [comment]
@aarishshahmohsin Please do not add reviewers, we have a system for this and must review the PR first. You are welcome to tag me to let me know it is ready.

**Comment by aarishshahmohsin** [comment]
> @aarishshahmohsin Please do not add reviewers, we have a system for this and must review the PR first. You are welcome to tag me to let me know it is ready. 

Ok I'll care of it from now on.

**Comment by Cioppolo14** [comment]
@anwersayeed @beingnoble03 Can you review this PR?

File: .github/workflows/check_branch.py
[the remark this answers is not in the document]
Comment (aarishshahmohsin) [reply .github/workflows/check_branch.py:1]: Should I change it?

**Comment by pranshugupta54** [comment]
LGTM 👍

**Review by pranshugupta54 — APPROVED** [review]

**Comment by palisadoes** [comment]
1. Please submit the same changes to a different branch to verify it fails. 
2. Link the PR here for evaluation

**Comment by aarishshahmohsin** [comment]
> 1. Please submit the same changes to a different branch to verify it fails.
> 
>     2. Link the PR here for evaluation
> 
>     3. This is also not a part of the `Code-Quality-Checks` section as requested in the issue

@palisadoes I have made the changes. 
The PR that checks for the case of the source and target branches being the same is [linked](https://github.com/PalisadoesFoundation/talawa-admin/actions/runs/8291690700/job/22691799336?pr=1728).

**Comment by beingnoble03** [comment]
@palisadoes not sure if this should be part of `Code-Quality-Checks`? i guess it should be part of `Check-Target-Branch`?

**Comment by palisadoes** [comment]
We are going to pause merging PRs into the `develop` branch to facilitate fixing the roles of Members, Admins and Super Admins.

They need to assigned on a per organization basis versus per API instance. This is the tracking issue.

- https://github.com/PalisadoesFoundation/talawa-api/issues/1965

Reviews of PRs for all branches will continue

**Comment by palisadoes** [comment]
> @palisadoes not sure if this should be part of `Code-Quality-Checks`? i guess it should be part of `Check-Target-Branch`?

My thought was to force failure if the branches didn't match. That's the better practice. Is this a valid approach in your opinion?

**Comment by beingnoble03** [comment]
> My thought was to force failure if the branches didn't match. That's the better practice. Is this a valid approach in your opinion?
yeah, that way it does look more appropriate to add it here.

**Review by beingnoble03 — APPROVED** [review]
lgtm

**Review by palisadoes — APPROVED** [review]

**Pull request merged**